Collect performance data during playbook runs

jladdjr opened the pull request:
Feature ticket:
https://github.com/ansible/tower/issues/3684

ryanpetrello commented:
@jladdjr I think `master` is the correct branch to target 1.4.x cc @matburt

ryanpetrello commented on packaging/rpm/ansible-runner.spec.j2:
```diff
@@ -47,6 +47,7 @@ Requires:       python-daemon
 %if 0%{?el7}
 Requires:       pexpect >= 4.6
 Requires:       python-psutil
+Requires:       libcgroup-tools
```
@matburt should we *require* this, or is the type of thing AWX should require downstream?  I imagine not *all* runner users will care about this feature.

matburt replied:
This should *not* be required by the package since it's an optional element.

ryanpetrello commented on ansible_runner/runner.py:
```diff
@@ -140,6 +144,18 @@ def run(self):
+            _, stderr = proc.communicate()
+            if proc.returncode:
+                # Unable to create cgroup, disable profiling
+                logger.warning('Unable to create cgroup: {}'.format(stderr))
```
Wondering if `logger.error` might make more sense.

matburt replied:
Yep, raise the severity of this.

ryanpetrello commented on ansible_runner/runner_config.py:
```diff
@@ -110,6 +112,11 @@ def __init__(self,
+        self.resource_profiling = resource_profiling
+        self.resource_profiling_base_cgroup = resource_profiling_base_cgroup
+        self.resource_profiling_cpu_poll_interval = resource_profiling_cpu_poll_interval
+        #TKTK: Update with options for mem, pid count
```
`TKTK` 😄 ?

jladdjr commented:
Updated docs:

- [x] Updated parameters in docs/intro.rst to match current parameters taken in the code
- [x] Provided more information on characteristics of what happens when perf stats is enabled
- [x] Info on how to set where data is collected, what data will be found there and what is the organization of the data
- [x] Concise example of what that data looks like.

jladdjr commented:
Also made another pass and made sure tests were current.

@matburt - this is ready for a final review.

matburt commented on ansible_runner/runner_config.py:
```diff
@@ -192,13 +202,38 @@ def prepare(self):
         if self.process_isolation:
             self.command = self.wrap_args_with_process_isolation(self.command)
 
+        if self.resource_profiling and self.execution_mode == ExecutionMode.ANSIBLE_PLAYBOOK:
```
Sorry, I didn't notice this from my previous reviews but why is this limited to just `ANSIBLE_PLAYBOOK`? Seems like it would be useful for ad-hoc mode also?  If not then maybe we should generate an error rather than silently not attaching the cgroup details.

jladdjr replied:
@matburt - 👍 Filed:
https://github.com/ansible/ansible-runner/issues/356
